### DISCUSSION TITLE
share .github/cpp-linter.yml in .github org repo

### DISCUSSION THREAD

**Discussion by shenxianpeng** [cpp-linter/cpp-linter#61] [Ideas]
This idea just comes to my mind if user wants to have a global configuration file `cpp-linter.yml` located in `.github` repo. (similar to [release-drafter.yml](https://github.com/cpp-linter/.github/blob/main/.github/release-drafter.yml) and [stale.yml](https://github.com/cpp-linter/.github/blob/main/.github/stale.yml))

I'm just here to record the idea.

**Comment by 2bndy5** [comment]
I'm not sure how feasible this is. We don't have support for a cpp-linter config file (as that is the use case for release-drafter and stalebot). Furthermore, to implement support for a config file that could be scoped to a GitHub organization (the .github repo), we would need to add that to cpp-linter-action logic.

Our cpp-linter-action's input are not that complex (though they are getting numerous), and this idea might be better served by a reusable workflow crafted by (and tailored to) a GitHub organization. I used this idea for the nRF24 org (see [nRF24's custom reusable workflow](https://github.com/nRF24/.github/blob/main/.github/workflows/cpp_lint.yaml)).

**Comment by shenxianpeng** [comment]
I like your implementation using reusable workflow 👍 this is also very elegant to achieve the same functionality.
